build-tools/packages/build-cli/src/commands/generate/sourceEntrypoints.ts

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { PackageJson } from "@fluidframework/build-tools";
+import { Flags } from "@oclif/core";
+import type { TsConfigJson } from "type-fest";
+import { generateSourceEntrypoints } from "../../library/commands/generateSourceEntrypoints.js";
+import { ApiLevel, ApiTag, BaseCommand } from "../../library/index.js";
+// AB#8118 tracks removing the barrel files and importing directly from the submodules, including disabling this rule.
+// eslint-disable-next-line import/no-internal-modules
+import { readPackageJson, readTsConfig } from "../../library/package.js";
+// AB#8118 tracks removing the barrel files and importing directly from the submodules, including disabling this rule.
+// eslint-disable-next-line import/no-internal-modules
+import { type ExportData, mapSourceExportsPath } from "../../library/packageExports.js";
+import type { CommandLogger } from "../../logging.js";
+
+/**
+ * Generates source entrypoints for Fluid Framework APIs to support API levels (/alpha, /beta. etc.).
+ */
+export default class GenerateSourceEntrypointsCommand extends BaseCommand<
+	typeof GenerateSourceEntrypointsCommand
+> {
+	static readonly description =
+		`Generates TypeScript source files that roll up APIs into different entrypoint files, defined by the "exports" field in package.json and organized by API tags`;
+
+	static readonly flags = {
+		mainEntrypoint: Flags.file({
+			description: "Main entrypoint file containing all untrimmed exports.",
+			default: "./src/index.ts",
+			exists: true,
+		}),
+		outDir: Flags.directory({
+			description: "Directory to emit entrypoint files.",
+			default: "./src/entrypoints/",
+			exists: true,
+		}),
+		...BaseCommand.flags,
+	};
+
+	public async run(): Promise<void> {
+		const { mainEntrypoint, outDir } = this.flags;
+
+		const packageJson = await readPackageJson();
+
+		const tsConfig = await readTsConfig();
+
+		const outFileSuffix = ".ts";
+		// Add a trailing slash if it is missing
+		const formattedOutDir = outDir.endsWith("/") ? outDir : `${outDir}/`;
+
+		const mapSrcQueryPathToApiTagLevel: Map<string | RegExp, ApiTag | undefined> = new Map([
+			[`${formattedOutDir}${ApiLevel.alpha}${outFileSuffix}`, ApiTag.alpha],
+			[`${formattedOutDir}${ApiLevel.beta}${outFileSuffix}`, ApiTag.beta],
+			[`${formattedOutDir}${ApiLevel.public}${outFileSuffix}`, ApiTag.public],
+			[`${formattedOutDir}${ApiLevel.legacy}${outFileSuffix}`, ApiTag.legacy],
+		]);
```

Comment:
Here or maybe in getOutputConfiguration we need to resolve source path to output .js/.d.ts mapping. One of the directions should be computable from `tsConfig`.
In the end you need map of `ApiTag` to source path (not to `ExportData`). `generateSourceEntrypoints` is only using `relPath` from `ExportData`; so, it should just be simplified to the path (source path).

Reply:
`generateSourceEntrypoints` uses `isTypeOnly` as well. I made a two minor changes to the function. 
1. Fixed the index.js path by moving up a directory.
2. Pass `isTypeOnly` (line 152)